modules/extension/brewer/src/main/java/org/geotools/styling/builder/ContrastEnhancementBuilder.java

```diff
@@ -49,18 +53,38 @@ public ContrastEnhancementBuilder gamma(double gamma) {
         return gamma(literal(gamma));
     }
 
-    public ContrastEnhancementBuilder normalize() {
-        this.method = ContrastMethod.NORMALIZE;
-        this.unset = false;
-        return this;
+    public ContrastEnhancementBuilder normalize(Map<String, Expression> constrastParameters) {
+        return contrastMethod("normalize", constrastParameters);
     }
 
-    public ContrastEnhancementBuilder histogram() {
-        this.method = ContrastMethod.HISTOGRAM;
-        this.unset = false;
-        return this;
+    public ContrastEnhancementBuilder histogram(Map<String, Expression> constrastParameters) {
+        return contrastMethod("histogram", constrastParameters);
+    }
+    public ContrastEnhancementBuilder exponential(Map<String, Expression> constrastParameters) {
+        return contrastMethod("exponential", constrastParameters);
     }
 
+    public ContrastEnhancementBuilder logarithmic(Map<String, Expression> constrastParameters) {
+        return contrastMethod("logarithm", constrastParameters);
+    }
+    private ContrastEnhancementBuilder contrastMethod(String name, Map<String, Expression> constrastParameters) {
+        if ("histogram".equals(name)) {
```

Comment:
This is stilly - make a method on the code list to look up by name ignoring case.